examples/binaries/constructor/src/lib.rs

```diff
@@ -0,0 +1,197 @@
+// This file is part of Gear.
+
+// Copyright (C) 2021-2022 Gear Technologies Inc.
+// SPDX-License-Identifier: GPL-3.0-or-later WITH Classpath-exception-2.0
+
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see <https://www.gnu.org/licenses/>.
+
+// This contract represents a general purpose contract that can perform a number
+// of actions in a loop, one action per loop iteration. Which of the 4 actions:
+// `Exec`, `Send` (to a program), `Send` (to a non-program) or `Trap` should be executed
+// is determined by some random value derived from the `handle` input.
+
+#![no_std]
+
+extern crate alloc;
+
+use alloc::{string::String, vec::Vec};
+
+#[cfg(feature = "std")]
+mod code {
+    include!(concat!(env!("OUT_DIR"), "/wasm_binary.rs"));
+}
+
+#[cfg(feature = "std")]
+pub use code::WASM_BINARY_OPT as WASM_BINARY;
+
+use parity_scale_codec::{Codec, Decode, Encode};
+
+pub type Key = String;
+
+#[derive(Clone, Debug, Decode, Encode)]
+pub enum Arg<T: 'static + Clone + Codec> {
+    New(T),
+    Get(Key),
+}
+
+impl<T: 'static + Clone + Codec> From<T> for Arg<T> {
+    fn from(value: T) -> Self {
+        Arg::New(value)
+    }
+}
+
+#[derive(Clone, Debug, Decode, Encode)]
+pub enum Call {
+    Store(Key),
+    StoreVec(Key),
+    Source,
+    Send(Arg<[u8; 32]>, Arg<Vec<u8>>, Option<u64>, u128, u32),
```

Comment:
Maybe better to use struct here?

Reply:
Can't see reason for it.
It much more comfortable to just write as fn args. Furthermore, there is Calls builder now, that is preferred to be used